### DISCUSSION TITLE
Session becomes `undefined` when passed to an external function inside field access control

### DISCUSSION THREAD

**Discussion by beerdedfellow** [keystonejs/keystone#8746] [Questions] [answered]
Hello Team,

I am currently stuck trying to implement field level access control. I have a very simple setup: I have an `Access` class in `lib/access/access.ts` that has a set of static functions assigned to it. I have been passing these functions directly like so:

```
        access: {
            operation: {
                query: Access.isSignedIn,
                create: Access.isSignedIn,
                update: Access.isSignedIn,
                delete: Access.isSuperAdmin,
            },
```
The functions are simple:

```
    static isSignedIn = ({ session }: { session: Session }) => {
        logger.debug(`isSignedIn Function: ${session?.data.id}`);
        return !!session?.data.id;
    };
...
    // Validate the current user is an Admin
    static isSuperAdmin = ({ session }: { session: any }) => {
        logger.debug(`isSuperAdmin Function session object: ${session}`);
        return !!session?.data.isSuperAdmin;
    };
```

Those seem to be working as I would expect, and all variables in the documentation are available to the function. However, I need to express something more complex, that is "OR", like so:

```
            first_name: text({
                validation: { isRequired: true },
                access: {
                    // only Admins or the owning Person can view name fields in the system
                    read: Access.isSuperAdmin || Access.isOrgAdmin || Access.isSelf
                    },
                },
            }),

```
However I can't express it that way, as read needs a function. So I put the anonymous function inline like so:

```
            first_name: text({
                validation: { isRequired: true },
                access: {
                    // only Admins or the owning Person can view name fields in the system
                    read: ({ session, context }) => {
                        logger.debug(`This is the session object passed to the read access control
                            on the first_name field: ${util.inspect(session)}`);
                        if (Access.isSuperAdmin(session)) return true;
                        // if (Access.isOrgAdmin(session)) return true;
                        // if (Access.isSelf(session, item)) return true;
                        return false;
                    },
                },
            }),
```

This is where things get weird. The `logger.debug` in the anonymous function logs a `session` object as I would expect. However the `logger.debug` inside of the `Access.isSuperAdmin` function is `undefined`. 

Is this an intended behavior that I am missing, or are we looking at a bug? I'm simply passing an argument to a function and then logging that value.This is replicating the behavior of the documentation: https://keystonejs.com/docs/config/access-control#field-access-control

Super grateful for any help you can provide!

**Comment by dcousens** [comment, marked as answer]
I think you wanted `({ session })` not `(session)`

```ts
if (Access.isSuperAdmin({ session })) return true
```
